header/store/store.go

```diff
@@ -332,7 +333,9 @@ func (s *store) flushLoop() {
 			continue
 		}
 
-		err := s.flush(s.pending.GetAll()...)
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+		defer cancel()
```

Comment:
What's the point of this defer? shouldn't you just call `cancel()` at the end of the loop statement bc we're creating a new context w timeout of 3 sec on every loop?